### ISSUE TITLE
Query with bernoulli tablesample fails in SliceDirectStreamReader

### ISSUE THREAD

**Issue by haozhun** [prestodb/presto#3563]
```
java.lang.IllegalArgumentException: Out of range: 2361557393
        at com.google.common.primitives.Ints.checkedCast(Ints.java:91)
        at com.facebook.presto.orc.reader.SliceDirectStreamReader.readBatch(SliceDirectStreamReader.java:108)
        at com.facebook.presto.orc.reader.SliceStreamReader.readBatch(SliceStreamReader.java:51)
        at com.facebook.presto.orc.OrcRecordReader.readVector(OrcRecordReader.java:279)
        at com.facebook.presto.hive.orc.OrcPageSource$LazySliceBlockLoader.load(OrcPageSource.java:473)
        at com.facebook.presto.hive.orc.OrcPageSource$LazySliceBlockLoader.load(OrcPageSource.java:453)
        at com.facebook.presto.spi.block.LazySliceArrayBlock.assureLoaded(LazySliceArrayBlock.java:180)
        at com.facebook.presto.spi.block.LazySliceArrayBlock.isEntryNull(LazySliceArrayBlock.java:96)
        at com.facebook.presto.spi.block.AbstractVariableWidthBlock.isNull(AbstractVariableWidthBlock.java:158)
        at com_facebook_presto_$gen_PageProcessor_5965.filter(Unknown Source)
        at com_facebook_presto_$gen_PageProcessor_5965.process(Unknown Source)
        at com.facebook.presto.operator.ScanFilterAndProjectOperator.getOutput(ScanFilterAndProjectOperator.java:232)
        at com.facebook.presto.operator.Driver.processInternal(Driver.java:380)
        at com.facebook.presto.operator.Driver.processFor(Driver.java:303)
        at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:587)
        at com.facebook.presto.execution.TaskExecutor$PrioritizedSplitRunner.process(TaskExecutor.java:505)
        at com.facebook.presto.execution.TaskExecutor$Runner.run(TaskExecutor.java:640)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by haozhun** [comment]
Rewriting the query from

    select count(*) from table TABLESAMPLE BERNOULLI (1) where ...

to

    select count(*) from table where ... and rand() < 0.01

also fails

**Comment by yuananf** [comment]
fixed by #3579
